spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ldap/embedded/EmbeddedLdapAutoConfiguration.java

```diff
@@ -107,6 +109,28 @@ public InMemoryDirectoryServer directoryServer() throws LDAPException {
 					this.embeddedProperties.getCredential().getUsername(),
 					this.embeddedProperties.getCredential().getPassword());
 		}
+
+		Schema schema = null;
+		if (this.embeddedProperties.getValidation().isEnabled()) {
+			schema = Schema.getDefaultStandardSchema();
+			String schemaLocation = this.embeddedProperties.getValidation().getSchema();
+			if (StringUtils.hasText(schemaLocation)) {
+				try {
+					Resource resource = this.applicationContext
+							.getResource(schemaLocation);
+					if (resource.exists()) {
+						schema = Schema.mergeSchemas(schema,
+								Schema.getSchema(resource.getFile()));
+					}
+				}
+				catch (Exception ex) {
+					throw new IllegalStateException(
+							"Unable to load schema " + schemaLocation, ex);
+				}
+			}
+		}
+		config.setSchema(schema);
```

Comment:
Not sure why you need to keep a handle to the schema there outside of the if block. I'd rather keep all that private (unless set it to `null` has some kind of purpose?)